modules/apps/frontend-taglib/frontend-taglib-clay/src/main/java/com/liferay/frontend/taglib/clay/servlet/taglib/UserCardTag.java

```diff
@@ -0,0 +1,584 @@
+
+		Boolean disabled = getDisabled();
+		Boolean selectable = getSelectable();
+
+		if ((selectable != null) && selectable) {
+			jspWriter.write("<div class=\"custom-control custom-checkbox\">");
+			jspWriter.write("<label>");
+			jspWriter.write("<input ");
+
+			if ((disabled != null) && disabled) {
+				jspWriter.write("disabled=\"disabled\" ");
+			}
+
+			String inputName = getInputName();
+
+			if (Validator.isNotNull(inputName)) {
+				jspWriter.write("name=\"" + inputName + "\" ");
+			}
+
+			String inputValue = getInputValue();
+
+			if (Validator.isNotNull(inputValue)) {
+				jspWriter.write("name=\"" + inputValue + "\" ");
+			}
+
+			Boolean selected = getSelected();
+
+			if ((selected != null) && selected) {
+				jspWriter.write("checked=\"checked\" ");
+			}
+
+			jspWriter.write("class=\"custom-control-input\" ");
+			jspWriter.write("type=\"checkbox\">");
+			jspWriter.write("<span class=\"custom-control-label\"></span>");
+		}
+
+		jspWriter.write("<div class=\"aspect-ratio-item-center-middle ");
+		jspWriter.write("card-type-asset-icon\">");
+
+		StickerTag stickerTag = new StickerTag();
+
+		stickerTag.setCssClass("sticker-user-icon");
+		stickerTag.setDisplayType(getUserColorClass());
+		stickerTag.setShape("circle");
+
+		String imageSrc = getImageSrc();
+
+		if (Validator.isNotNull(imageSrc)) {
+			stickerTag.setContent(
+				StringBundler.concat(
+					"<img alt=\"", getImageAlt(),
+					"\" class=\"aspect-ratio-item-fluid\" src=\"", imageSrc,
+					"\" />"));
+		}
+		else {
+			String icon = getIcon();
+
+			if (icon != null) {
+				stickerTag.setIcon(icon);
+			}
+			else {
+				stickerTag.setIcon("user");
+			}
+		}
+
+		stickerTag.doTag(pageContext);
+
+		jspWriter.write("</div>");
+
+		if ((selectable != null) && selectable) {
+			jspWriter.write("</label>");
+			jspWriter.write("</div>");
+		}
+
+		jspWriter.write("</div>");
+
+		jspWriter.write("<div class=\"card-body\">");
+		jspWriter.write("<div class=\"card-row\">");
+		jspWriter.write("<div class=\"autofit-col autofit-col-expand\">");
+		jspWriter.write("<p class=\"card-title\">");
+		jspWriter.write("<span class=\"text-truncate-inline\">");
+
+		String href = getHref();
+		String name = getName();
+
+		if (((disabled == null) || !disabled) && Validator.isNotNull(href)) {
+			jspWriter.write("<a class=\"text-truncate\" href=\"");
+			jspWriter.write(href);
+			jspWriter.write("\">");
+			jspWriter.write(name);
+			jspWriter.write("</a>");
+		}
+		else {
+			jspWriter.write("<span class=\"text-truncate\">");
+			jspWriter.write(name);
+			jspWriter.write("</span>");
+		}
+
+		jspWriter.write("</span>");
+		jspWriter.write("</p>");
+		jspWriter.write("<p class=\"card-subtitle\">");
+		jspWriter.write("<span class=\"text-truncate-inline\">");
+		jspWriter.write("<span class=\"text-truncate\">");
+		jspWriter.write(getSubtitle());
+		jspWriter.write("</span>");
+		jspWriter.write("</span>");
+		jspWriter.write("</p>");
```

Comment:
Are we allowed to have these nice writes, or we need to uglify code and merge strings to 80 chars?

Reply:
Why?

Reply:
Because Brian asked nicely enough :)

He wanted a simple algorithm for this which is mostly used everywhere. When passing a string, you put as much as you can until a `"` or space or `=` and then break to new lines when it exceeds 80 chars.

Java expressions take their own line, so the code above should look something like:

```java
		jspWriter.write("<p class=\"card-subtitle\"><span class=\"");
		jspWriter.write("text-truncate-inline\"><span class=\"");
		jspWriter.write("text-truncate\">");
		jspWriter.write(getSubtitle());
		jspWriter.write("</span></span></p>");
```

You should be able to see it in the other tags.

Reply:
Because Brian says so it's enough, but I don't like it. It's hard enough having the clay markup duplicated here to also having it "offuscated" in that way that will be pretty hard to understand what's going on. I mean, nobody likes debuggin on compiled code right? The same here.

Reply:
Believe me... I know...

Reply:
I believe you, but I had to say it, it's pretty frustrating. There's no way we can change his mind?

Reply:
You can never say never... but I don't think this fight is worth the energy

Reply:
Plus you'd have to change all code following this pattern in portal, which is quite a lot...

Reply:
Regardless of how hard it is to debug, it is true that is simpler to enforce because the rules are way simpler

Reply:
It's important to "pick your battles" so that you save your fighting resources for the things that really matter. In this case, it might be annoying, but the big picture is that _we don't want to be in the business of regularly writing out HTML by string concatenation in Java_. In other words, we expect this kind of work to be very occasional in nature (like when we need to create or port Clay tags, which should not be often), and spend most of our time writing JSX in JS files. Same goes for the pain of writing JS in JSP: we could make it less painful, or we could try to structure things so that we have fewer reasons to write JS in JSP in the first place.

Reply:
I'll add a commit with the message `Uglify markup rendering so we can suffer when debugging`

Reply:
We probably want to squash most if not all commits before forwarding this 😉